com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/verifier/JarVerifier.java

```diff
@@ -0,0 +1,193 @@
+package com.sap.cloud.lm.sl.cf.process.util.verifier;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.JarURLConnection;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.ResourceAccessException;
+
+import com.google.common.base.Splitter;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+
+@Component
+public class JarVerifier implements Verifier {
+
+    private static final String META_INF = "META-INF";
+    private static final String CN = "CN";
+    private static final int BUFFER_SIZE = 8 * 1024;
+
+    @Override
+    public void verify(List<X509Certificate> targetCertificates, String certificateCN, URL jarURL) {
+        try {
+            JarFile jarFile = retrieveJarFile(jarURL);
+            List<JarEntry> jarEntries = getJarEntries(jarFile);
+            verifyJarEntries(targetCertificates, jarEntries, certificateCN);
+        } catch (IOException e) {
+            throw new ResourceAccessException(Messages.CERTIFICATE_VERIFICATION_HAS_FAILED, e);
+        }
+    }
+
+    private JarFile retrieveJarFile(URL jarURL) throws IOException {
+        jarURL = new URL("jar:" + jarURL.toString() + "!/");
+        JarURLConnection connection = (JarURLConnection) jarURL.openConnection();
+        return connection.getJarFile();
+    }
+
+    private List<JarEntry> getJarEntries(JarFile jarFile) throws IOException {
+        List<JarEntry> jarEntries = new ArrayList<>();
+        Enumeration<JarEntry> entries = jarFile.entries();
+        while (entries.hasMoreElements()) {
+            readJarEntry(jarFile, jarEntries, entries);
+        }
+        return jarEntries;
+    }
+
+    private void readJarEntry(JarFile jarFile, List<JarEntry> jarEntries, Enumeration<JarEntry> entries) throws IOException {
+        JarEntry jarEntry = entries.nextElement();
+        validateJarEntryPath(jarFile, jarEntry);
+        if (jarEntry.isDirectory()) {
+            return;
+        }
+        jarEntries.add(jarEntry);
```

Comment:
Return the `jarEntry` and add it to the list in the `getJarEntries` method. That way you'll reduce the number of arguments given to this method by 1.

Reply:
Done.